hs_core/tests/api/views/test_authorize.py

```diff
@@ -0,0 +1,853 @@
+                      # private resource (both metadata and content files
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.VIEW_RESOURCE,
+                       'success': False, 'exception': PermissionDenied}
+                      ] + common_parameters
+
+        self._run_tests(self.request, parameters)
+
+        # test for discoverable resource
+        self.res.raccess.discoverable = True
+        self.res.raccess.public = False
+        self.res.raccess.save()
+
+        parameters = [
+                      # authenticated user (with no assigned access permission) has no authorization for resource view
+                      # (metadata and content files) for a discoverable resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.VIEW_RESOURCE,
+                       'success': False, 'exception': PermissionDenied},
+                      # authenticated (with no assigned access permission) user has authorization for resource metadata
+                      # only view for a discoverable resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.VIEW_METADATA,
+                       'success': True, 'exception': None}
+                      ] + common_parameters
+
+        self._run_tests(self.request, parameters)
+
+        # test for public resource
+        self.assertTrue(self.res.raccess.discoverable)
+        self.assertFalse(self.res.raccess.public)
+        self.res.raccess.discoverable = False
+        self.res.raccess.public = True
+        self.res.raccess.save()
+
+        parameters = [
+                      # authenticated user (with no assigned access permission) has authorization for resource view
+                      # (metadata and content files) for a public or published resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.VIEW_RESOURCE,
+                       'success': True, 'exception': None},
+                      # authenticated user (with no assigned access permission)has authorization for resource metadata
+                      # only view for a public or published resource
+                      {'res_id': self.res.short_id, 'needed_permission': Action_To_Authorize.VIEW_METADATA,
+                       'success': True, 'exception': None}
+                      ] + common_parameters
+
+        self._run_tests(self.request, parameters)
+
+        # test for immutable/published resource
+        self.assertFalse(self.res.raccess.immutable)
+        self.assertFalse(self.res.raccess.published)
+        self.assertTrue(self.res.raccess.public)
+        self.res.raccess.immutable = True
+        self.res.raccess.published = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+    def test_authorize_inactive_user(self):
```

Comment:
@hyi As I understand the web login checks for active status for authentication. However, the is_authenticated() function does not check for that status. That means for an inactive user user.is_authenticated() will return True. If we don't do user active status check it would be a problem for REST API.